src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -392,6 +392,395 @@ namespace OpenMS
+ *
+ * @param[in,out] mz_rt_ranges
+ *   A vector of pairs of `RangeMZ` and `RangeRT` specifying the m/z and RT ranges over which to aggregate data.
+ *   Each pair defines a rectangular region in the m/z-RT plane. The vector will be sorted in-place by ascending
+ *   minimum m/z and descending maximum m/z within the function.
+ *
+ * @param[in] ms_level
+ *   The MS level of the spectra to be processed. Only spectra matching this MS level will be considered in the aggregation.
+ *
+ * @param[in] func_mz_reduction
+ *   A function or functor that performs the m/z reduction. It should have the signature:
+ *   `CoordinateType func_mz_reduction(MSSpectrum::ConstIterator begin_it, MSSpectrum::ConstIterator end_it);`
+ *   The function receives iterators to the beginning and end of the peaks within the m/z range for a specific spectrum.
+ *   It should process these peaks and return a single `CoordinateType` value representing the aggregated result.
+ *
+ * @return
+ *   A vector of vectors of `MSExperiment::CoordinateType`. Each sub-vector corresponds to an m/z and RT range in
+ *   `mz_rt_ranges`, and contains the aggregated results for each spectrum that falls within that RT range.
+ *   The size of each sub-vector equals the number of spectra that overlap with the RT range.
+ *
+ * @note
+ * - If `mz_rt_ranges` is empty or there are no spectra at the specified MS level, the function returns an empty vector.
+ * - The `mz_rt_ranges` vector will be sorted within the function by ascending minimum m/z and descending maximum m/z.
+ * - The function uses OpenMP for parallelization over spectra. Ensure that your reduction function is thread-safe.
+ * - The aggregation is performed only on the peaks that fall within both the specified m/z and RT ranges.
+ * - This methods works best with larger number of m/z and RT ranges and a large number of spectra.
+ *
+ * @warning
+ * - The function modifies `mz_rt_ranges` by sorting it. If the original order is important, make a copy before calling.
+ * - The provided `func_mz_reduction` must be able to handle empty ranges (i.e., when `begin_it == end_it`).
+ *
+ * @exception None
+ */
+template<class MzReductionFunctionType>
+std::vector<std::vector<MSExperiment::CoordinateType>> aggregate(
+    std::vector<std::pair<RangeMZ, RangeRT>>& mz_rt_ranges,
+    unsigned int ms_level,
+    MzReductionFunctionType func_mz_reduction) const
+{
+    // Early exit if there are no ranges
+    if (mz_rt_ranges.empty()) 
+    {
+      // likely an error, but we return an empty vector instead of throwing an exception for now
+      return {};
+    }
+
+    // Sort mz_rt_ranges by ascending MinMZ and descending MaxMZ
+    std::sort(mz_rt_ranges.begin(), mz_rt_ranges.end(), 
+        [](const auto& a, const auto& b) 
+        {
+          if (a.first.getMinMZ() != b.first.getMinMZ()) 
+          {
+            return a.first.getMinMZ() < b.first.getMinMZ();
+          }
+          return -a.first.getMaxMZ() < -b.first.getMaxMZ();
```

Comment:
This might be useful elsewhere.

Add `bool RangeBase::operator<(const RangeBase& rhs)`?

Use
```suggestion
          return std::tie(getMinMZ(), getMaxMZ()) < std::tie(rhs.getMinMZ(), rhs.getMaxMZ());
```
there?

And then you can simply call sort() without lambda.

Reply:
tie needs lvalue unfortunately

Reply:
ah you mean in RangeBase?

Reply:
Hmm for RangeBase: Would you always want to prefer bigger ranges? (i.e. descending max?). This might be specific to this use case. But I would not mind in general.

Reply:
hmm I think it is better to keep it out of rangebase. might be unexpected